libs/message-coder/README.md

```diff
@@ -0,0 +1,67 @@
+# `message-coder`
+
+Provides a simple way to encode and decode binary messages from/to JavaScript
+objects.
+
+## Usage
+
+```ts
+import {
+  CoderType,
+  fixedString,
+  literal,
+  message,
+  uint8,
+} from '@votingworks/message-coder';
+
+enum Genre {
+  Blues = 0x00,
+  ClassicRock = 0x01,
+  Country = 0x02,
+  Dance = 0x03,
+  Disco = 0x04,
+  Funk = 0x05,
+  Grunge = 0x06,
+  HipHop = 0x07,
+  Jazz = 0x08,
+  // ...
+}
+
+// define a coder for ID3v1 tags
+const id3v1 = message({
+  header: literal('TAG'),
+  title: fixedString(30),
+  artist: fixedString(30),
+  album: fixedString(30),
+  year: fixedString(4),
+  comment: fixedString(28),
+  zeroByte: literal(0x00),
+  track: uint8(),
+  genre: uint8(Genre),
+});
+type id3v1 = CoderType<typeof id3v1>;
```

Comment:
Some useful things to know about this:

1. `message` coders know how to ignore `literal` and `padding` coder properties, so the inferred type of `id3v1` is `MessageCoder<{ title: string; artist: string; album: string; year: string; comment: string; track: number; genre: Genre }>`, which does not include `header` or `zeroByte`.
2. `CoderType` extracts the value that a given `Coder` type encodes (similar to `z.TypeOf` in zod), and that type is given a name that matches the coder value as a convention that I think I like.

Reply:
I think that'd be fine, though there may be some benefit in keeping them the same. So that'd be `Id3v1` for both, which I'm fine with.